.github/workflows/test.yaml

```diff
@@ -0,0 +1,33 @@
+name: Test
+on: [push, pull_request]
+jobs:
+  test_linux:
+    timeout-minutes: 20
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        clion_version: [2019.3, 2020.1, 2020.2]
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+      - uses: actions/setup-python@v1
+      - name: Get Conan
+        run: pip install conan &&
+          conan profile new default --detect &&
+          conan profile update settings.compiler.libcxx=libstdc++11 default
+      - name: Cache downloads
```

Comment:
This step actually downloads the cache. There is a post-run step this step runs to actually upload the cache. This is why this step must be before running the tests (which gets dependencies and builds as well).